manifests/init.pp

```diff
@@ -154,11 +170,15 @@
   Systemd::LogindSettings                                $logind_settings,
   Boolean                                                $manage_all_network_files,
   Stdlib::Absolutepath                                   $network_path,
-  Hash                                                   $loginctl_users = {},
-  Hash                                                   $dropin_files = {},
-  Hash                                                   $udev_rules = {},
+  Hash[String,Hash[String, Any]]                         $loginctl_users,
```

Comment:
Could you please add spec tests for these? Here's an example at https://github.com/ghoneycutt/puppet-module-bind/blob/master/spec/classes/init_spec.rb#L511-L534 

You specify params with a valid hash and then check to see that the defined types are in the catalog.